components/brave_news/browser/suggestions_controller.cc

```diff
@@ -272,50 +273,62 @@ SuggestionsController::GetSuggestedPublisherIdsWithHistory(
-                         api_request_helper::APIRequestResult
-                             api_request_result) {
-                        controller->locale_ = locale;
-                        controller->similarities_ = ParseSimilarityResponse(
-                            api_request_result.TakeBody(), locale);
-                        controller->on_current_update_complete_->Signal();
-                        controller->is_update_in_progress_ = false;
-                        controller->on_current_update_complete_ =
-                            std::make_unique<base::OneShotEvent>();
-                      },
-                      base::Unretained(controller), locale),
-                  {}, {.auto_retry_on_network_change = true});
-            },
-            base::Unretained(controller), locale));
-      },
-      base::Unretained(this)));
+  publishers_controller_->GetLocale(
+      subscriptions,
+      base::BindOnce(
+          [](SuggestionsController* controller,
+             const BraveNewsSubscriptions& subscriptions,
+             const std::string& locale) {
+            controller->publishers_controller_->GetOrFetchPublishers(
+                subscriptions,
+                base::BindOnce(
+                    [](SuggestionsController* controller,
+                       const std::string& locale, Publishers publishers) {
+                      const GURL url(
+                          "https://" + brave_news::GetHostname() +
+                          "/source-suggestions/source_similarity_t10." +
+                          locale + ".json");
+                      controller->api_request_helper_->Request(
+                          "GET", url, "", "",
+                          base::BindOnce(
+                              [](SuggestionsController* controller,
+                                 std::string locale,
+                                 api_request_helper::APIRequestResult
+                                     api_request_result) {
+                                controller->locale_ = locale;
+                                controller->similarities_ =
+                                    ParseSimilarityResponse(
+                                        api_request_result.TakeBody(), locale);
+                                controller->on_current_update_complete_
+                                    ->Signal();
+                                controller->is_update_in_progress_ = false;
+                                controller->on_current_update_complete_ =
+                                    std::make_unique<base::OneShotEvent>();
+                              },
+                              base::Unretained(controller), locale),
+                          {}, {.auto_retry_on_network_change = true});
+                    },
+                    base::Unretained(controller), locale));
+          },
+          base::Unretained(this), subscriptions));
```

Comment:
This seems to be so-called callback hell in JS World 😄 I'm curious if there's something we can do

Reply:
...we can figure out how to use `await` in C++ 😆...
But yes maybe for these smaller classes (and when refactoring to smaller classes elsewhere), we can use slightly fewer inline lambdas. But it does seem in this PR we're at least not adding more layers, just re-formatting?

Reply:
Yeah, just reformatting, unfortunately. C++ callback hell is real :D